garrysmod/lua/includes/modules/worldtip.lua

```diff
@@ -0,0 +1,56 @@
+
+local IsValid = IsValid
+local SysTime = SysTime
+local isvector = isvector
+
+
+/*
+	Name: worldtip
+	Desc: Handles world/entity-based tooltips
+*/
+module( "worldtip" )
+
+local WorldTips = {}
+
+function GetTable()
+	return WorldTips
+end
+
+function Add( name, tgt, text, len, col )
+
+	local ent, pos = IsValid( tgt ) && tgt
```

Comment:
I blame github for a terribad UI that doesn't convey to the user what is happening.

Anyway, I recommend changing it to:
```lua
local pos
local ent = IsValid( tgt ) && tgt
```
Simply because it's a lot more readable. That trick is confusing as hell. Even if you disagree, please remember that most gmod coders are going to be Lua newbies and some _might_ read over this code. Best not to confuse them as it confused me.